server/src/weaverbird/backends/mongo_translator/utils.py

```diff
@@ -1,7 +1,93 @@
     return f'vqb_{col_name_without_invalid_chars}'
+
+
+class UnsupportedOperatorError(Exception):
+    """Raised when condition uses an unsupported operator"""
+
+
+def build_cond_expression(
+    cond: Union[SimpleCondition, ConditionComboOr, ConditionComboAnd]
+) -> MongoStep:
+    operator_mapping = {
+        'eq': '$eq',
+        'ne': '$ne',
+        'lt': '$lt',
+        'le': '$lte',
+        'gt': '$gt',
+        'ge': '$gte',
+        'in': '$in',
+        'nin': '$in',
+        'isnull': '$eq',
+        'notnull': '$ne',
+        'matches': '$regexMatch',
+        'notmatches': '$regexMatch',
+        'from': '$gte',
+        'until': '$lte',
+    }
+    unsupported_operators: List = []
+
+    if isinstance(cond, ConditionComboAnd):
+        if len(cond.and_) == 1:
+            return build_cond_expression(cond.and_[0])
+        else:
+            return {'$and': [build_cond_expression(elem) for elem in cond.and_]}
+    if isinstance(cond, ConditionComboOr):
+        return {'$and': [build_cond_expression(elem) for elem in cond.or_]}
+    if cond.operator in unsupported_operators:
+        raise UnsupportedOperatorError(f'Unsupported operator ${cond.operator} in conditions')
+
+    if cond.operator == 'matches' or cond.operator == 'notmatches':
+        cond_expression: MongoStep = {
+            '$regexMatch': {'input': f'${cond.column}', 'regex': cond.value}
+        }
+        if cond.operator == 'notmatches':
+            cond_expression = {'$not': cond_expression}
+        return cond_expression
+
+    else:
+        if cond.operator == 'notnull' or cond.operator == 'isnull':
+            return {operator_mapping[cond.operator]: [f'${cond.column}', None]}
+
+        else:
+            cond_expression = {operator_mapping[cond.operator]: [f'${cond.column}', cond.value]}
+            if cond.operator == 'nin':
+                cond_expression = {'$not': cond_expression}
+
+            if cond.operator == 'until':
+                if isinstance(cond.value, datetime.datetime):
+                    cond_expression[operator_mapping[cond.operator]][1] = [
+                        datetime.datetime(
+                            day=cond.value.day, month=cond.value.month, year=cond.value.month
+                        ).replace(hour=23, minute=59, second=59, microsecond=999999)
+                    ]
+
+            if cond.operator == 'from' or cond.operator == 'until':
+                return {
+                    '$expr': {
+                        operator_mapping[cond.operator]: [
+                            truncate_to_day(f'${cond.column}'),
+                            truncate_to_day(
+                                translate_relative_date(cond.value)
+                                if isinstance(cond.value, RelativeDate)
+                                else cond.value
+                            ),
+                        ]
+                    }
+                }
+            return cond_expression
```

Comment:
can we make it simplify here ?
i would suggest splitting in little function here.